pages/company/visualizations/company_associated_activity.py

```diff
@@ -0,0 +1,253 @@
+            clearable=True,
+        ),
+    )
+    return date_picker
+
+
+# callback for Company Affiliation by Github Account Info graph
+@callback(
+    Output(VIZ_ID, "figure"),
+    [
+        Input("repo-choices", "data"),
+        Input(f"{PAGE}-{paramter_2}-{VIZ_ID}", "value"),
+        Input(f"{PAGE}-{paramter_1}-{VIZ_ID}", "value"),
+        Input(f"{PAGE}-date-picker-range-{VIZ_ID}", "start_date"),
+        Input(f"{PAGE}-date-picker-range-{VIZ_ID}", "end_date"),
+    ],
+    background=True,
+    prevent_initial_call=True,
+)
+def compay_associated_activity_graph(repolist, checks, num, start_date, end_date):
+
+    # wait for data to asynchronously download and become available.
+    cache = cm()
+    df = cache.grabm(func=cq, repos=repolist)
+    while df is None:
+        time.sleep(1.0)
+        df = cache.grabm(func=cq, repos=repolist)
+
+    start = time.perf_counter()
+    logging.debug(f"{VIZ_ID}- START")
+
+    # test if there is data
+    if df.empty:
+        logging.debug(f"{VIZ_ID} - NO DATA AVAILABLE")
+        return nodata_graph
+
+    # function for all data pre processing, COULD HAVE ADDITIONAL INPUTS AND OUTPUTS
+    df = process_data(df, checks, num, start_date, end_date)
+
+    fig = create_figure(df)
+
+    logging.debug(f"{VIZ_ID} - END - {time.perf_counter() - start}")
+    return fig
+
+
+def process_data(df: pd.DataFrame, checks, num, start_date, end_date):
+
+    # convert to datetime objects rather than strings
+    df["created"] = pd.to_datetime(df["created"], utc=True)
+
+    # order values chronologically by COLUMN_TO_SORT_BY date
+    df = df.sort_values(by="created", axis=0, ascending=True)
+
+    # filter values based on date picker
+    if start_date is not None:
+        df = df[df.created >= start_date]
+    if end_date is not None:
+        df = df[df.created <= end_date]
+
+    # creates list of emails for each contribution and flattens list result
+    emails = df.email_list.str.split(" , ").explode("email_list").tolist()
+
+    # remove any entries not in email format
+    emails = [x for x in emails if "@" in x]
+
+    # creates list of email domains from the emails list
+    email_domains = [x[x.rindex("@") + 1 :] for x in emails]
+
+    # creates df of domains and counts
+    df = pd.DataFrame(email_domains, columns=["domains"]).value_counts().to_frame().reset_index()
+
+    df = df.rename(columns={0: "occurances"})
+
+    # changes the name of the company if under a certain threshold
+    df.loc[df.occurances <= num, "domains"] = "Other"
+
+    # groups others together for final counts
+    df = (
+        df.groupby(by="domains")["occurances"]
+        .sum()
+        .reset_index()
+        .sort_values(by=["occurances"], ascending=False)
+        .reset_index(drop=True)
+    )
+
+    # removes entries with gmail or other if checked
+    if "gmail" in checks:
+        df = df[df.domains != "gmail.com"]
+    if "other" in checks:
+        df = df[df.domains != "Other"]
+
+    return df
```

Comment:
this processing logic makes 100% sense to me, need to check whether the company query might cause a multi-count of contributors but not sure.

Reply:
So this visualization is counting contributions not contributors so it will have multiple counts for contributors